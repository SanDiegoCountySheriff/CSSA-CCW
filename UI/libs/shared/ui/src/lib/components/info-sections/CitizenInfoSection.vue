<template>
  <v-container class="citizen-info-container rounded">
    <v-banner class="font-weight-bold text-xl text-left mb-5">
      {{ $t('Citizenship Information: ') }}
    </v-banner>
    <v-row class="ml-5">
      <v-col>
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('Citizen:') }}
          </v-chip>
          <div class="info-text">
            {{ props.citizenshipInfo.citizen }}
          </div>
        </div>
      </v-col>
      <v-col>
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t(' Military Status:') }}
          </v-chip>
          <div class="info-text">
            {{ props.citizenshipInfo.militaryStatus }}
          </div>
        </div>
      </v-col>
    </v-row>

    <v-row class="ml-5">
      <v-col>
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('Immigrant Alien') }}
          </v-chip>
          <div class="info-text">
            {{ props.immigrantInfo.immigrantAlien }}
          </div>
        </div>
      </v-col>
      <v-col>
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t(' Non Immigrant Alien ') }}
          </v-chip>
          <div class="info-text">
            {{ props.immigrantInfo.nonImmigrantAlien }}
          </div>
        </div>
      </v-col>
    </v-row>
    <v-row class="ml-5">
      <v-col>
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t(' Country of Birth ') }}
          </v-chip>
          <div class="info-text">
            {{ props.immigrantInfo.countryOfBirth }}
          </div>
        </div>
      </v-col>
      <v-col>
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t(' Country of Citizenship ') }}
          </v-chip>
          <div class="info-text">
            {{ props.immigrantInfo.countryOfCitizenship }}
          </div>
        </div>
      </v-col>
    </v-row>
  </v-container>
</template>

<script setup lang="ts">
import {
  CitizenshipType,
  ImmigrantInformation,
} from '@shared-utils/types/defaultTypes';

interface ICitizenShipInfoSectionProps {
  citizenshipInfo: CitizenshipType;
  immigrantInfo: ImmigrantInformation;
  color: string;
}

const props = defineProps<ICitizenShipInfoSectionProps>();
</script>

<style lang="scss">
.citizen-info-container {
  width: 80%;
  height: 100%;
  margin: 0;
  padding: 0;
}
.info-row {
  display: flex;
  flex-direction: row;
  margin-left: 0.5rem;
}

.info-text {
  margin-left: 0.5rem;
  text-align: start;
  height: 1.8em;
  width: 50%;
  margin-bottom: 0.5rem;
  padding-left: 0.5rem;
  padding-top: 0.2rem;
  background-color: rgba(211, 241, 241, 0.3);
  border-bottom: 1px solid #666;
  border-radius: 5px;
  font-size: 1.2em;
  font-weight: bold;
}
</style>
