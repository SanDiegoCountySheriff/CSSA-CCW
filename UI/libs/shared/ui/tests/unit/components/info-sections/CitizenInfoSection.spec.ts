/* eslint-disable @typescript-eslint/ban-ts-comment */
import CitizenInfoSection from '@shared-ui/components/info-sections/CitizenInfoSection.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();

describe('CitizenInfoSection', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    //@ts-ignore
    wrapper = mount(CitizenInfoSection, {
      localVue,
      vuetify,
      mocks: {
        $t: text => text,
      },
      propsData: {
        citizenshipInfo: {
          citizen: true,
          militaryStatus: 'active',
        },
        immigrantInfo: {},
      },
    });
  });

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
