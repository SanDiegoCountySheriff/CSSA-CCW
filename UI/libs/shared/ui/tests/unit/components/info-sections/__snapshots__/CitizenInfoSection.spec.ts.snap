// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitizenInfoSection Should match the snapshot 1`] = `
<div class="container citizen-info-container rounded">
  <transition-stub name="expand-transition" class="font-weight-bold text-xl text-left mb-5">
    <div class="v-banner v-sheet theme--light v-banner--is-mobile">
      <div class="v-banner__wrapper">
        <div class="v-banner__content">
          <div class="v-banner__text">
            Citizenship Information:
          </div>
        </div>
      </div>
    </div>
  </transition-stub>
  <div class="row ml-5">
    <div class="col">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          Citizen:
        </span></span>
        <div class="info-text">
          true
        </div>
      </div>
    </div>
    <div class="col">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
           Military Status:
        </span></span>
        <div class="info-text">
          active
        </div>
      </div>
    </div>
  </div>
  <div class="row ml-5">
    <div class="col">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          Immigrant Alien
        </span></span>
        <div class="info-text">

        </div>
      </div>
    </div>
    <div class="col">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
           Non Immigrant Alien 
        </span></span>
        <div class="info-text">

        </div>
      </div>
    </div>
  </div>
  <div class="row ml-5">
    <div class="col">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
           Country of Birth 
        </span></span>
        <div class="info-text">

        </div>
      </div>
    </div>
    <div class="col">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
           Country of Citizenship 
        </span></span>
        <div class="info-text">

        </div>
      </div>
    </div>
  </div>
</div>
`;
